ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -34,9 +35,14 @@ import (
 	"github.com/urfave/cli"
 )
 
+// user data builder can be easily mocked in tests
+var newUserDataBuilder func(string) userdata.UserDataBuilder = userdata.NewBuilder
```

Comment:
For this kind of thing, I would probably use an interface rather than a global var, but I'm also not sure why this needs to be mocked in the test.

Reply:
I created this mock for use in  `TestClusterUpWithUserData`, which tests that cluster up correctly sets the user data parameter in the CFN template. But you're right... the test could be re-written to not need to mock this function.